### PR TITLE
Migrate `epic.interactivity` to Playwright

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Migrate `epic.interactivity` to Playwright

This test required updating the `waitForIsland` util to accept a `waitFor` option that allows us to expect an island to be `attached` | `detached` | `visible` | `hidden` .

This is because the island `LiveBlogEpic` is slightly strange in that it does not become visible and inserts the Epic component itself imperatively.

### CODE REVIEW

File: dotcom-rendering/playwright/lib/islands.ts [hunk 1]
```diff
@@ -4,36 +4,34 @@ type IslandStatus = 'rendered' | 'hydrated';
 type WaitForIslandOptions = {
 	status?: IslandStatus;
 	nth?: number;
+	waitFor?: 'attached' | 'detached' | 'visible' | 'hidden';
```
Comment (mxdvl) [inline dotcom-rendering/playwright/lib/islands.ts:7]: What about `state`?

File: dotcom-rendering/playwright/lib/islands.ts [hunk 2]
```diff
@@ -4,36 +4,34 @@ type IslandStatus = 'rendered' | 'hydrated';
...
 	const islandLocator = page.locator(islandSelector).nth(nth);
 	// check that the island is present on the page
-	await islandLocator.isVisible();
+	// await islandLocator.isVisible();
+	await islandLocator.waitFor({ state: 'attached', timeout: 30_000 });
 	// scroll to it
 	await islandLocator.scrollIntoViewIfNeeded();
 	// wait for it to be rendered or hydrated
-	const hyrdatedIslandSelector = `gu-island[name="${island}"][data-island-status="${status}"]`;
+	const hyrdatedIslandSelector = `gu-island[name="${name}"][data-island-status="${status}"]`;
```
Comment (mxdvl) [inline dotcom-rendering/playwright/lib/islands.ts:31]: `hyrdatedIslandSelector` &rarr; `hydratedIslandSelector`

File: dotcom-rendering/playwright/lib/islands.ts [hunk 3]
```diff
@@ -4,36 +4,34 @@ type IslandStatus = 'rendered' | 'hydrated';
...
 ): Promise<void> => {
-	const { status = 'rendered', nth = 0 } = options;
-	const islandSelector = `gu-island[name="${island}"]`;
+	const { status = 'rendered', nth = 0, waitFor = 'visible' } = options;
+	const islandSelector = `gu-island[name="${name}"]`;
 	// create a locator for the island
 	const islandLocator = page.locator(islandSelector).nth(nth);
 	// check that the island is present on the page
-	await islandLocator.isVisible();
+	// await islandLocator.isVisible();
```
Comment (mxdvl) [inline dotcom-rendering/playwright/lib/islands.ts:26]:
```suggestion
```

**Review by mxdvl — APPROVED** [review]
🎭

File: dotcom-rendering/playwright/lib/islands.ts (hunk 1 above)
Comment (arelra) [reply dotcom-rendering/playwright/lib/islands.ts:7]: Might be confusing with the island 'status'.

Could do `waitForState`

Comment (mxdvl) [reply dotcom-rendering/playwright/lib/islands.ts:7]: Status is going away very soon: #allislands

**Pull request merged**